classes/PrestashopConfiguration.php

```diff
@@ -0,0 +1,157 @@
+                array(
+                    'fopen' => (ConfigurationTest::test_fopen() || ConfigurationTest::test_curl()),
+                    'admin_au_writable' => ConfigurationTest::test_dir($this->autoupgradeDir, false),
+                    'shop_deactivated' => (!Configuration::get('PS_SHOP_ENABLE') || (isset($_SERVER['HTTP_HOST']) && in_array($_SERVER['HTTP_HOST'], array('127.0.0.1', 'localhost')))),
+                    'cache_deactivated' => !(defined('_PS_CACHE_ENABLED_') && _PS_CACHE_ENABLED_),
+                    'module_version_ok' => $this->checkAutoupgradeLastVersion($this->getUpgrader()->autoupgrade_last_version)
+            ));
+        }
+        return $this->allowed_array;
+    }
+
+    /**
+     * @return string|false Returns the module version if found in the config.xml file, false otherwise.
+     */
+    public function getModuleVersion()
+    {
+        if (!is_null($this->moduleVersion)) {
+            return $this->moduleVersion;
+        }
+
+        // TODO: to be moved as property class in order to make tests possible
+        $path = _PS_ROOT_DIR_.'/modules/autoupgrade/config.xml';
+
+        $this->moduleVersion = false;
+        if (file_exists($path)
+            && $xml_module_version = simplexml_load_file($path)
+        ) {
+            $this->moduleVersion = (string)$xml_module_version->version;
+        }
+        return $this->moduleVersion;
+    }
+
+    public function getPrestaShopVersion()
+    {
+        if (defined('_PS_VERSION_')) {
+            return _PS_VERSION_;
+        }
+        $files = array(
+            $this->psRootDir.'/config/settings.inc.php',
+            $this->psRootDir.'/config/autoload.php',
+        );
+        foreach ($files as $file) {
+            $version = $this->findPrestaShopVersionInFile(file_get_contents($file));
+            if ($version) {
+                return $version;
+            }
+        }
+
+        throw new \Exception('Can\'t find PrestaShop Version');
+    }
+
+    /**
+     * Compares the installed module version with the one available on download
+     * 
+     * @return boolean True is the latest version of the module is currently installed
+     */
+    public function checkAutoupgradeLastVersion($extAutoupgradeLastVersion)
+    {
+        $moduleVersion = $this->getModuleVersion();
+        if ($moduleVersion) {
+            return version_compare($moduleVersion, $extAutoupgradeLastVersion, '>=');
+        }
+        return true;
+    }
+
+    /**
+     * @return array Details of the filesystem permission check
+     */
+    protected function getRootWritableDetails()
+    {
+        $result = array();
+        // Root directory permissions cannot be checked recursively anymore, it takes too much time
+        $result['root_writable'] =  ConfigurationTest::test_dir('/', false, $report);
+        $result['root_writable_report'] = $report ? $report : true; // Avoid null in the array as it makes the shop non-compliant
```

Comment:
If report can be a boolean or null, maybe cast it in boolean instead using ternary?

Reply:
Hehe, the variable is `null` if no error occurred, and containsa string if an error needs to be reported.